feat(vite-plugin): Add Containers-related info logs

CarmenPopoviciu opened the pull request:
Fixes DEVX-2052

Adds logs, when a Worker has Containers configured, providing information about container build status, and how to rebuild containers during local development.

`dev`

<img width="986" height="426" alt="Screenshot 2025-07-29 at 14 26 59" src="https://github.com/user-attachments/assets/3ac3ad7e-5888-42e1-8a3b-31d7909d32b9" />

`preview`

<img width="1045" height="425" alt="Screenshot 2025-07-29 at 14 30 56" src="https://github.com/user-attachments/assets/f5dfc46b-b71f-44da-ade1-61aa0f163df2" />

---

<!--
Please don't delete the checkboxes <3
The following selections do not need to be completed if this PR only contains changes to .md files
-->

- Tests
  - [ ] Tests included
  - [x] Tests not necessary because: just a log change
- Public documentation
  - [ ] Cloudflare docs PR(s): <!--e.g. <https://github.com/cloudflare/cloudflare-docs/pull/>...-->
  - [x] Documentation not necessary because: we're keeping docs lite on this sort of details
- Wrangler V3 Backport
  - [ ] Wrangler PR: <!--e.g. <https://github.com/cloudflare/workers-sdk/pull/>...-->
  - [x] Not necessary because: containers is a beta feat that only exists in v4

<!--
Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?
In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.
-->

jamesopstad commented on packages/vite-plugin-cloudflare/src/index.ts:
```diff
@@ -449,13 +449,19 @@ if (import.meta.hot) {
 					}
 
 					if (hasDevContainers) {
+						viteDevServer.config.logger.warn(
+							"Building container images for local development..."
+						);
```
```suggestion
						viteDevServer.config.logger.info(
							"Building container images for local development..."
						);
```

Unless there was a reason you chose `warn`?

jamesopstad commented on packages/vite-plugin-cloudflare/src/index.ts:
```diff
@@ -449,13 +449,19 @@ if (import.meta.hot) {
 							dockerPath,
 							configPath: entryWorkerConfig.configPath,
 						});
+						viteDevServer.config.logger.warn(
+							"Containers succesfully built. To rebuild your containers during development, restart the Vite dev server (r + enter)."
+						);
```
```suggestion
						viteDevServer.config.logger.info(
							"Containers successfully built. To rebuild your containers during development, restart the Vite dev server (r + enter)."
						);
```

CarmenPopoviciu replied:
same

jamesopstad reviewed (commented):
I would have thought `info` is more appropriate than `warn` but interested to know your thoughts.

Should we also add similar logs to preview?

It might be nice to add some colour to the logging (we haven't been very consistent about this but it's one area to improve). I'll let you decide what's best, but you can `import colors from "picocolors";` (like [here](https://github.com/cloudflare/workers-sdk/blob/c070c7799b0edbf33610a3da68f738da7bc78405/packages/vite-plugin-cloudflare/src/build.ts#L107-L114)).

CarmenPopoviciu commented:
> I would have thought `info` is more appropriate than `warn` but interested to know your thoughts.
> 
> Should we also add similar logs to preview?
> 
> It might be nice to add some colour to the logging (we haven't been very consistent about this but it's one area to improve). I'll let you decide what's best, but you can `import colors from "picocolors";` (like [here](https://github.com/cloudflare/workers-sdk/blob/c070c7799b0edbf33610a3da68f738da7bc78405/packages/vite-plugin-cloudflare/src/build.ts#L107-L114)).

My apologies for the unintended `warn` use. I think I did something funky when autocompleting. Log level was meant to be `info` indeed. Thank you for spotting that!

Indeed, some color would be nice, since the message gets lost in the sea of log output. Juts wasn't sure if we have any convention. I'll have a look

I'll add something for preview as well 👍

jamesopstad commented:
> Indeed, some color would be nice, since the message gets lost in the sea of log output. Juts wasn't sure if we have any convention. I'll have a look

I think the convention is basically, yellow (like [here](https://github.com/cloudflare/workers-sdk/blob/c070c7799b0edbf33610a3da68f738da7bc78405/packages/vite-plugin-cloudflare/src/miniflare-options.ts#L215-L221)) 😅. Definitely room for improvement though.

CarmenPopoviciu commented:
@jamesopstad I addressed your comments in https://github.com/cloudflare/workers-sdk/pull/10071/commits/9b9abfdf21ef4b610263716283654c8f3e873beb and updated the screenshots in the PT description

jamesopstad reviewed (approved):
Thanks @CarmenPopoviciu!
